packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue

```diff
@@ -22,48 +22,62 @@
       mode="click"
       padding-size="small"
       close-on-click
+      @show-drop="updateAppIcons"
     >
-      <oc-list class="applications-list">
-        <li v-for="(n, nid) in applicationsList" :key="`apps-menu-${nid}`" @click="clickApp(n)">
-          <oc-button
-            :key="n.url ? 'apps-menu-external-link' : 'apps-menu-internal-link'"
-            :type="n.url ? 'a' : 'router-link'"
-            :target="n.target"
-            :href="n.url"
-            :to="n.path"
-            :appearance="n.active ? 'raw-inverse' : 'raw'"
-            :variation="n.active ? 'primary' : 'passive'"
-            :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
-          >
-            <span class="icon-box">
-              <oc-icon :name="n.icon" variation="inherit" />
-            </span>
-            <span v-text="$gettext(n.title)" />
-            <oc-icon v-if="n.active" name="check" class="active-check" variation="inherit" />
-          </oc-button>
-        </li>
-      </oc-list>
+      <div class="oc-display-block oc-position-relative">
+        <oc-list class="applications-list">
+          <li v-for="(n, nid) in applicationsList" :key="`apps-menu-${nid}`">
+            <oc-button
+              :key="n.url ? 'apps-menu-external-link' : 'apps-menu-internal-link'"
+              :type="n.url ? 'a' : 'router-link'"
+              :target="n.target"
+              :href="n.url"
+              :to="n.path"
+              :appearance="n.active ? 'raw-inverse' : 'raw'"
+              :variation="n.active ? 'primary' : 'passive'"
+              :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
+            >
+              <oc-application-icon :key="appIconKey" :icon="n.icon" :color-primary="n.color" />
+              <span v-text="$gettext(n.title)" />
+            </oc-button>
+          </li>
+        </oc-list>
+      </div>
     </oc-drop>
   </nav>
 </template>
 
 <script lang="ts">
-import { defineComponent, PropType, ComponentPublicInstance } from 'vue'
-import { configurationManager } from 'web-pkg/src/configuration'
-import { urlJoin } from 'web-client/src/utils'
+import { defineComponent, PropType, ComponentPublicInstance, ref, unref, computed } from 'vue'
 import { OcDrop } from 'design-system/src/components'
+import OcApplicationIcon from 'design-system/src/components/OcApplicationIcon/OcApplicationIcon.vue'
+import { useGettext } from 'vue3-gettext'
 
 export default defineComponent({
+  components: {
+    OcApplicationIcon
+  },
   props: {
     applicationsList: {
       type: Array as PropType<any[]>,
       required: false,
       default: () => []
     }
   },
-  computed: {
-    applicationSwitcherLabel() {
-      return this.$gettext('Application Switcher')
+  setup() {
+    const { $gettext } = useGettext()
+    const appIconKey = ref(false)
```

Comment:
According to the Vue docs, child keys should be unique. You can have a look here e.g.: https://github.com/owncloud/web/blob/redesign-apps-menu/packages/web-app-files/src/components/AppBar/CreateAndUpload.vue#L297

It simply creates a random prefix and uses it with the loop index in https://github.com/owncloud/web/blob/redesign-apps-menu/packages/web-app-files/src/components/AppBar/CreateAndUpload.vue#L112.